src/impls.rs

```diff
@@ -47,67 +47,67 @@ impl<T: Config> Pallet<T> {
 			.unwrap_or_default()
 	}
 
-	/// Executes the vote.
+	/// Executes the vote and transfers corresponding dispute fees to initiator/reporter.
 	/// # Arguments
 	/// * `dispute_id` - The identifier of the dispute.
-	/// * `vote_round` - The vote round.
 	/// * `result` - The result of the dispute, as determined by governance.
-	pub(super) fn execute_vote(
-		dispute_id: DisputeId,
-		vote_round: u8,
-		result: VoteResult,
-	) -> DispatchResult {
-		// Ensure validity of id
+	pub(super) fn execute_vote(dispute_id: DisputeId, result: VoteResult) -> DispatchResult {
+		// Ensure validity of dispute id, vote has been executed, and vote must be tallied
 		ensure!(
 			dispute_id != <DisputeId>::default() &&
 				dispute_id != Keccak256::hash(&[]) &&
 				<DisputeInfo<T>>::contains_key(dispute_id),
 			Error::<T>::InvalidDispute
 		);
-
-		for round in (1..=vote_round).rev() {
-			<VoteInfo<T>>::try_mutate(dispute_id, round, |maybe| -> DispatchResult {
-				match maybe {
-					None => Err(Error::<T>::InvalidVote.into()),
-					Some(vote) => {
-						// Ensure vote has not already been executed, and vote must be tallied
-						ensure!(!vote.executed, Error::<T>::VoteAlreadyExecuted);
-						ensure!(vote.tally_date > 0, Error::<T>::VoteNotTallied);
-						// Ensure vote must be final vote and that time has to be pass after the vote is tallied
-						if round == vote_round {
-							ensure!(
-								<VoteRounds<T>>::get(vote.identifier) == vote.vote_round,
-								Error::<T>::VoteNotFinal
-							);
+		let final_vote_round = <VoteRounds<T>>::get(dispute_id);
+		ensure!(final_vote_round > 0, Error::<T>::InvalidVote);
+		<VoteInfo<T>>::try_mutate(dispute_id, final_vote_round, |maybe| -> DispatchResult {
+			match maybe {
+				None => Err(Error::<T>::InvalidVote.into()),
+				Some(vote) => {
+					// Ensure vote has not already been executed, and vote must be tallied
+					ensure!(!vote.executed, Error::<T>::VoteAlreadyExecuted);
+					ensure!(vote.tally_date > 0, Error::<T>::VoteNotTallied);
+					// Ensure that time has to be passed after the vote is tallied (86,400 = 24 * 60 * 60 for seconds in a day)
+					ensure!(
+						Self::now().saturating_sub(vote.tally_date) >= 1 * DAYS,
+						Error::<T>::TallyDisputePeriodActive
+					);
+					vote.executed = true;
+					vote.result = Some(result);
+					let dispute =
+						<DisputeInfo<T>>::get(dispute_id).ok_or(Error::<T>::InvalidDispute)?;
+					<OpenDisputesOnId<T>>::mutate(dispute.query_id, |maybe| {
+						if let Some(disputes) = maybe {
+							disputes.saturating_dec();
```

Comment:
shouldn't we decrement it for each dispute_rounds?

Reply:
Double-checked the usages of `OpenDisputesOnId` and it is only incremented on the first round of a dispute (query id, timestamp): https://github.com/tellor-io/tellor-pallet/blob/6a1fe70a722f4038c9c49701d5406ae9baf650c4/src/lib.rs#L1091-L1099

Reply:
yes, my bad.

Just curious to know, is there any reason for incrementing it on the first round only?

Reply:
The main usage of `OpenDisputesOnId` is in calculating the dispute fee, where the fee increases based on the number of open disputes: https://github.com/tellor-io/tellor-pallet/blob/6a1fe70a722f4038c9c49701d5406ae9baf650c4/src/lib.rs#L1100-L1108

I guess technically speaking the dispute is on the data (query id and timestamp more specifically), so subsequent rounds of voting are still disputing the same data but I think you are right to clarify it. @oraclown could you provide additional context on this please, especially after the refactoring to align the dispute_id with your parachain implementation? 🙂 

As it stands, if it isn't being incremented per vote round then the choice of a u8 for vote round may not be sufficient?

Reply:
Its actually the vote round which is a `u8` rather than open disputes (which is a `u128`), based on the previous assumption that each vote round would require an increasing dispute fee, thereby limiting the rounds.

The current implementation will max out at `u8::MAX`, but it is perhaps better to add an explicit error such as `MaxVoteRoundsReached` once this has been reached? 

This could always be updated to a larger type in the future, but will obviously require a runtime upgrade along with storage migration as necessary.

Reply:
Upon reflection I thought it best to return an error once max vote rounds reached so have implemented it along with a test. I will merge once CI completes as successful.